### PR TITLE
cleanup: remove kube-proxy-images.json

### PR DESCRIPTION
**Pull request by jason1028kr**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
remove kube-proxy-image.json
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/components.json
Comment (cameronmeissner) [inline parts/linux/cloud-init/artifacts/components.json:386]: doesn't seem like we're actually using this during VHD builds (install-dependencies.sh), are we not supposed to be caching this anymore?

Comment (cameronmeissner) [reply parts/linux/cloud-init/artifacts/components.json:386]: confirmed offline that we still aim to cache this, should probably wait on merging this until after @Devinwong's change merge into master

Comment (jason1028kr) [reply parts/linux/cloud-init/artifacts/components.json:386]: Wouldn't this loop https://github.com/Azure/AgentBaker/pull/4556/files#diff-d6aeb8e845af4c32251915a17dc7c887d73cd628262e83118b9dcd65f7728c98L312-R343 cover the added json entry under 'containerImages' key? I couldn't properly check b/c test pipeline was failing for other reasons, but I'll confirm once that's unblocked. I can also just pause until schema's finalized.

Comment (cameronmeissner) [reply parts/linux/cloud-init/artifacts/components.json:386]: ah, yes you're right :)

approved, up to you whether you wanna merge now before Devin's changes or wait until the schema is updated

File: parts/linux/cloud-init/artifacts/components.json
Comment (djsly) [inline parts/linux/cloud-init/artifacts/components.json:389]: that's way too may version. after we review the Schema, we will need to keep 3 images only (one per AKS versions)

Comment (jason1028kr) [reply parts/linux/cloud-init/artifacts/components.json:390]: I believe we'll need to keep corresponding kube-proxy version for each unique major.minor.patch k8s version

**Review by Devinwong — APPROVED** [review]

**Pull request merged**
